modules/apps/frontend-js/frontend-js-react-web/src/main/java/com/liferay/frontend/js/react/web/internal/importmap/FrontendJsReactWebJSImportmapsContributor.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.react.web.internal.importmap;
+
+import com.liferay.frontend.js.importmaps.extender.JSImportmapsContributor;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+
+import javax.servlet.ServletContext;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(immediate = true, service = JSImportmapsContributor.class)
+public class FrontendJsReactWebJSImportmapsContributor
+	implements JSImportmapsContributor {
+
+	@Override
+	public JSONObject getImportmapsJSONObject() {
+		return _importmapsJSONObject;
+	}
+
+	@Activate
+	protected void activate() {
+		_importmapsJSONObject = _jsonFactory.createJSONObject();
+
+		String contextPath = _servletContext.getContextPath();
+
+		for (String moduleName : _MODULE_NAMES) {
+			_importmapsJSONObject.put(
+				moduleName,
+				StringBundler.concat(
+					contextPath, "/__liferay__/amd2esm/", moduleName, ".js"));
+		}
+	}
+
+	private static final String[] _MODULE_NAMES = {
+		"classnames", "formik", "prop-types", "react", "react-dnd",
+		"react-dnd-html5-backend", "react-dom"
+	};
```

Comment:
Will this be necessary for every module that contributes esm modules? Seems a little annoying to have to specify these modules both in our .java file and npmscripts config

Reply:
Yes, for now. As I said somewhere (in a commit, I think) we want to make this complex for now, to prevent an abuse of the mechanism. The more modules we export, the more the importmap grows and there's no way back, once we have published something as an ESM module.

So, better be very very conservative with these exports :grimacing: .

Reply:
Here's the explanation -> https://github.com/liferay-frontend/liferay-portal/pull/1883/commits/193388078cb3f00d0b7735d499bb3fa70b538ff0

Reply:
Ah, yes okay I remember that now. Thanks for the clarification!